strmangle.go

```diff
@@ -17,9 +17,9 @@ var (
 	idAlphabet    = []byte("abcdefghijklmnopqrstuvwxyz")
 	smartQuoteRgx = regexp.MustCompile(`^(?i)"?[a-z_][_a-z0-9\-]*"?(\."?[_a-z][_a-z0-9]*"?)*(\.\*)?$`)
 
-	rgxEnum            = regexp.MustCompile(`^enum(\.[a-zA-Z0-9_]+)?\((,?'[^']+')+\)$`)
-	rgxEnumIsOK        = regexp.MustCompile(`^(?i)[a-z][a-z0-9_\s]*$`)
-	rgxEnumShouldTitle = regexp.MustCompile(`^[a-z][a-zA-Z0-9_]*$`)
+	rgxEnum            = regexp.MustCompile(`^enum(\.[a-zA-Z0-9_.:]+)?\((,?'[^']+')+\)$`)
+	rgxEnumIsOK        = regexp.MustCompile(`^(?i)[a-z][a-z0-9_.:\s]*$`)
+	rgxEnumShouldTitle = regexp.MustCompile(`^[a-z][a-zA-Z0-9_.:]*$`)
```

Comment:
Since enum values are "any valid string", I suspect we will continue to run into issues with edgecases considering our approach here. Perhaps it should be different?

Reply:
I reckon these will become obsolete if we replace all non-alphanumeric characters as you suggested. I don't think we _need_ to make a decision at the moment though. It doesn't seem like a widespread problem if I'm the only one who have (allegedly) encountered these constraints.

Reply:
If I remember correctly, these are used to check if the enum is valid before trying to title case it or anything. so without it, it wouldn't get to other parts.